app/DynamicGraphs/GraphGenerator.hs

```diff
@@ -7,34 +7,110 @@ module DynamicGraphs.GraphGenerator
 import Data.GraphViz.Attributes as A
 import Data.GraphViz.Attributes.Complete as AC
 import Data.GraphViz.Types.Generalised (
+  DotEdge(..),
   DotGraph(..),
   DotNode(..),
   DotStatement(..),
   GlobalAttributes(..))
 import Database.Requirement (Req(..))
 import Data.Sequence as Seq
-import Data.Text.Lazy (Text)
+import Data.Text.Lazy (Text, pack)
 
 
+-- Serves as a sort of "interface" for the whole part "dynamic graph"
 sampleGraph :: DotGraph Text
 sampleGraph = reqsToGraph [
-    ("CSC108", NONE)
+    ("CSC148", J "CSC108")
     ]
 
 
 -- ** Main algorithm for converting requirements into a graph
+
+-- (not so sure) reqToStmts are meant to convert a single requirement and reqsToGraph use concatMap to 
+-- use reqToStmts to converts a list of requirements all at once and concatenate the results into a 
+-- single list of DotGraph objects.
 reqsToGraph :: [(Text, Req)] -> DotGraph Text
 reqsToGraph reqs =
-    let stmts = concatMap reqToStmts reqs
+    let stmts = stmtslst (foldUpReqLst reqs 0)
     in
         buildGraph stmts
 
-reqToStmts :: (Text, Req) -> [DotStatement Text]
-reqToStmts (name, NONE) = [DN $ DotNode name []]
-reqToStmts (_, _) = []
+-- convert the original requirement data into dot statements that can be used by buildGraph to create the
+-- corresponding DotGraph objects. ([] is the [] after name the optional parameters for DotNode?)
+reqToStmts :: (Text, Req) -> StmtsWithCounter -> StmtsWithCounter
```

Comment:
Since this is now a helper function for `foldl`, you can flip the order of the parameters here. That way, you'll be able to just call `foldl reqToStmts`.

Reply:
I am sorry I do not quite get it, would you mind giving some extra advise on how to adjust the design to fit the 
reqToStmts into foldl that way? Thanks!

Reply:
Below, you have a call to `foldl` that looks like `foldl (\x y -> f y x)`. If you reorder the parameters of `f`, then instead you could write `foldl (\x y -> f x y)`, which is simplified to `foldl f`.